### PR TITLE
Tracing playground new data model adjustments

### PR DESCRIPTION
**Pull request by monsterkrampe**
This PR contains two features regarding tracing (that we also have for version `0.4.0` on the `tracing-playground` branch):
- read facts to be traced from file
- output GraphML format for trace result (required by Evonne)

### CODE REVIEW

**Comment by monsterkrampe** [comment]
(I will rebase this again once the build on main has been fixed.)

File: nemo-cli/src/cli.rs [hunk 1]
```diff
@@ -119,10 +119,13 @@ impl OutputArgs {
 pub(crate) struct TracingArgs {
     /// Facts for which a derivation trace should be computed;
     /// multiple facts can be separated by a semicolon
-    #[arg(long = "trace", value_delimiter = ';')]
-    pub(crate) traced_facts: Option<Vec<String>>,
+    #[arg(long = "trace", value_delimiter = ';', group = "trace-input")]
+    pub(crate) facts_to_be_traced: Option<Vec<String>>,
+    /// Specify an input file for the facts that should be explained
```
Comment (aannleax) [inline nemo-cli/src/cli.rs:124]: I would give a hint here about the expected input format

File: nemo-cli/src/main.rs [hunk 2]
```diff
@@ -170,11 +170,41 @@ fn run(mut cli: CliApp) -> Result<(), Error> {
...
+                        .filter_map(|filename| {
+                            match read_to_string(filename.clone()).map_err(|err| {
+                                ReadingError::IoReading {
+                                    error: err,
+                                    filename: filename.to_string_lossy().to_string(),
+                                }
+                            }) {
+                                Ok(inner) => Some(inner),
+                                Err(err) => {
+                                    log::warn!("!Warning: {err}");
```
Comment (aannleax) [inline nemo-cli/src/main.rs:189]: This should probably result in an error

File: nemo/src/execution/tracing/trace.rs [hunk 3]
```diff
@@ -182,7 +187,36 @@ pub struct TraceTreeRuleApplication {
     _position: usize,
 }
 
-/// Tree representation of an [ExecutionTrace] from a given start node
+impl TraceTreeRuleApplication {
+    fn to_instantiated_rule(&self) -> Rule {
```
Comment (aannleax) [inline nemo/src/execution/tracing/trace.rs:191]: Would be nice to have a doc comment also for the private methods.

**Review by aannleax — CHANGES_REQUESTED** [review]
Very cool :)

Have some minor comments

**Review by aannleax — APPROVED** [review]
Looks good, thanks :)

**Pull request merged**